src/main/java/com/mcmoddev/mmdbot/modules/logging/misc/ScamDetector.java

```diff
@@ -0,0 +1,130 @@
+
+    public static final List<String> SCAM_LINKS = Collections.synchronizedList(new ArrayList<>());
+
+    static {
+        new Thread(ScamDetector::setupScamLinks, "Scam link collector").start();
+    }
+
+    @Override
+    public void onGuildMessageReceived(@NotNull final GuildMessageReceivedEvent event) {
+        final var msg = event.getMessage();
+        if (containsScam(msg)) {
+            final var member = msg.getMember();
+            final var guild = msg.getGuild();
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            final var embed = getLoggingEmbed(msg, "");
+            msg.delete().queue($ -> {
+                getLoggingChannel(guild).sendMessageEmbeds(embed).queue();
+                // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+                guild.addRoleToMember(member, guild.getRoleById(mutedRoleID)).queue();
+            });
+        }
+    }
+
+    @Override
+    public void onGuildMessageUpdate(@NotNull final GuildMessageUpdateEvent event) {
+        final var msg = event.getMessage();
+        if (containsScam(msg)) {
+            final var member = msg.getMember();
+            final var msgChannel = msg.getTextChannel();
+            final var guild = msg.getGuild();
+            final var mutedRoleID = MMDBot.getConfig().getRole("muted");
+            final var embed = getLoggingEmbed(msg, ", by editing an old message");
+            msg.delete().queue($ -> {
+                getLoggingChannel(guild).sendMessageEmbeds(embed).queue();
+                // TODO once JDA is updated, maybe timeout the user. It seems a better idea
+                guild.addRoleToMember(member, guild.getRoleById(mutedRoleID)).queue();
+            });
+        }
+    }
+
+    private static MessageEmbed getLoggingEmbed(final Message message, final String extraDescription) {
+        final var member = message.getMember();
+        return new EmbedBuilder().setTitle("Scam link detected!")
+            .setDescription(String.format("User %s sent a scam link in %s%s! Their message was deleted, and they were muted!", member.getAsMention(),
+                message.getTextChannel().getAsMention(), extraDescription))
+            .addField("Message Content", "> " + message.getContentRaw(), false)
+            .setColor(Color.RED)
+            .setTimestamp(Instant.now())
+            .setFooter("User ID: " + member.getIdLong())
+            .setThumbnail(member.getEffectiveAvatarUrl()).build();
+    }
+
+    private static TextChannel getLoggingChannel(final Guild guild) {
+        return guild.getTextChannelById(MMDBot.getConfig().getChannel("events.requests_deletion"));
+    }
+
+    public static boolean containsScam(final Message message) {
+        final String msgContent = message.getContentRaw().toLowerCase();
+        for (final var link : SCAM_LINKS) {
+            if (msgContent.contains(link)) {
+                return true;
+            }
+        }
```

Comment:
The [javadocs for `Collections.synchronizedList`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/util/Collections.html#synchronizedList(java.util.List)) note that when using an `Iterator` on the returned list (as what enhanced `for`-loops do), it must be manually synchronized:
> It is imperative that the user manually synchronize on the returned list when traversing it via `Iterator`, `Spliterator` or `Stream`:
> ```java
> List list = Collections.synchronizedList(new ArrayList());
>     ...
> synchronized (list) {
>     Iterator i = list.iterator(); // Must be in synchronized block
>     while (i.hasNext())
>         foo(i.next());
> }
> ```
> Failure to follow this advice may result in non-deterministic behavior.

Additionally, we may want to include pathways to heuristically detect if a message is likely not a scam to avoid the cost of processing _every_ message. One thing I want to see is to exclude messages from non-users (the bot user itself, possibly other bot users, and other things like webhooks or Discord messages).

Other examples: 
 - If the user is above or equal the bot in roles (essentially, Moderators and Admins), they should be exempt as trusted users. 
 - Preprocess the list of domains, collect all the top level domains, and check that (hopefully) smaller set first before testing against the whole domain list.